### ISSUE TITLE
Document How to Enable Hot Reloading Using dotnet run/watch

### ISSUE THREAD

**Issue by RehanSaeed** [dotnet/aspnetcore#38036]
### Is your feature request related to a problem? Please describe.
It is not clear how to enable hot reloading using `dotnet run` or `dotnet watch` from a `launchSettings.json` file. The .NET [blog](https://devblogs.microsoft.com/aspnet/asp-net-core-updates-in-net-6-preview-3/#initial-net-hot-reload-support) mention that we should add the `"hotReloadProfile": "aspnetcore"` setting like so:

```json
"Kestrel": {
  "commandName": "Project",
  "dotnetRunMessages": true,
  "hotReloadProfile": "aspnetcore"
  "launchBrowser": true,
  "launchUrl": "weatherforecast",
  "applicationUrl": "http://localhost:5093",
  "environmentVariables": {
    "ASPNETCORE_ENVIRONMENT": "Development"
  }
},
```

However, I also came across this PR which removes this need:

- https://github.com/dotnet/aspnetcore/pull/32469/files

Then there is the question about how to enable the feature when using `dotnet watch` in a `launchSettings.json` file:

```
"Watch Kestrel (CTRL+F5)": {
  "commandName": "Executable",
  "executablePath": "dotnet",
  "workingDirectory": "$(ProjectDir)",
  "commandLineArgs": "watch",
  "environmentVariables": {
    "ASPNETCORE_ENVIRONMENT": "Development"
  }
},
```

It would also be nice to know if this affects IIS Express or WSL2

```json
    "IIS Express": {
      "commandName": "IISExpress",
      "launchBrowser": true,
      "launchUrl": "graphql",
      "environmentVariables": {
        "ASPNETCORE_ENVIRONMENT": "Development"
      }
    },
    "WSL 2": {
      "commandName": "WSL2",
      "launchBrowser": true,
      "launchUrl": "http:////localhost:5000",
      "environmentVariables": {
        "ASPNETCORE_URLS": "http://localhost:5000",
        "ASPNETCORE_ENVIRONMENT": "Development"
      },
      "distributionName": ""
    },
```

### Describe the solution you'd like
I'd like to know how to enable the feature for both `dotnet run` and `dotnet watch` from a `launchSettings.json` file and have this documented somewhere.

**Comment by mkArtakMSFT** [comment]
Thanks for contacting us.
Looks like @taha-ghadirian answered this already on how to achieve what you're looking for. Closing, as there are no more actions pending here.

**Comment by Drit-team** [comment]
Hello @mkArtakMSFT, I think we should consider it as an a bug, because normally we just need to run `dotnet watch` command without `--project .` flag when we are in the same directory of the project.

Also `dotnet watch` command will fail when system goes to sleep. So we should restart system (Not restarting dotnet watch) to fix this issue.

**Comment by taha-ghadirian** [comment]
Hello @mkArtakMSFT, I think we should consider it as an a bug, because normally we just need to run dotnet watch command without --project . flag when we are in the same directory of the project.

Also dotnet watch command will fail when system goes to sleep. So we should restart system (Not restarting dotnet watch) to fix this issue.

**Comment by chhinsras** [comment]
I use dotnet watch run or dotnet run inside Server project (Blazor Hosted). File changed detected but No hot reload change to apply , why?

**Comment by taha-ghadirian** [comment]
> I use dotnet watch run or dotnet run inside Server project (Blazor Hosted). File changed detected but No hot reload change to apply , why?

Actually it's a bug in `dotnet watch` command, please run this command in `Server` project and use `--project .` flag. 
This solution will fix that issue.

**Comment by chhinsras** [comment]
> > I use dotnet watch run or dotnet run inside Server project (Blazor Hosted). File changed detected but No hot reload change to apply , why?
> 
> Actually it's a bug in `dotnet watch` command, please run this command in `Server` project and use `--project .` flag. This solution will fix that issue.

are you saying go to Server folder. then run dotnet watch --project ? will it run Client project as well?

**Comment by taha-ghadirian** [comment]
Hello @chhinsras, Yes!
Open Server project, run `Dotnet watch --project .` or `dotnet watch --project .\ProjectName.csproj` or `dotnet watch --project ProjectName.csproj` one of this commands should work.

**Comment by taha-ghadirian** [comment]
> > > I use dotnet watch run or dotnet run inside Server project (Blazor Hosted). File changed detected but No hot reload change to apply , why?
> > 
> > 
> > Actually it's a bug in `dotnet watch` command, please run this command in `Server` project and use `--project .` flag. This solution will fix that issue.
> 
> are you saying go to Server folder. then run dotnet watch --project ? will it run Client project as well?
> 
> ![image](https://user-images.githubusercontent.com/31751665/140886685-5e7df3a1-1c49-41ae-b321-82997fe9e47e.png) I followed and NO client project wont have hot reload

Can you run it again with `-v` flag? Then copy all logs for us, thanks :)

**Comment by chhinsras** [comment]
> > are you saying go to Server folder. then run dotnet watch --project ? will it run Client project as well?
> > ![image](https://user-images.githubusercontent.com/31751665/140886685-5e7df3a1-1c49-41ae-b321-82997fe9e47e.png) I followed and NO client project wont have hot reload
> 
> Can you run it again with `-v` flag? Then copy all logs for us, thanks :)

![image](https://user-images.githubusercontent.com/31751665/140887571-b5f2af11-b3ca-498d-9356-97bd7c258c3f.png)

**Comment by taha-ghadirian** [comment]
> > > are you saying go to Server folder. then run dotnet watch --project ? will it run Client project as well?

Look at last line, it says: `Watch: Hot reload capabilities: .` it should say: `Watch: Hot reload capabilities: BASELINE.`.

please use this comment then check if it's fixed or not: `dotnet watch --project .\Server.csproj -v`
If it wouldn't fix, try changing your default browser to a new chrome browser (without any extension installed).

**Comment by jhulbertpmn** [comment]
Same issue here. 

info: Microsoft.AspNetCore.Hosting.Diagnostics[1]
      Request starting HTTP/2 POST https://localhost:5001/_framework/blazor-hotreload application/json 2
watch : Received 1 from browser in [Count: 1, MessageType: Binary, EndOfMessage: True].
watch : No hot reload changes to apply.
watch : Hot reload change handled in 839.5826ms.

**Comment by chhinsras** [comment]
> in any workaround

**Comment by ceeol** [comment]
Tried all the different suggestions, and this was really a struggle.
I did manage to get hot reload working by creating the small template project by using exactly --project .\Server.csproj, and getting Baseline.
No luck with another project though, just getting "watch : Hot reload capabilities: ." This was earlier created with .Net 5, and I discovered there were upgrades. Finally got it to work. It needs to be more user friendly than this, though.
